### PR TITLE
Remove f_luc* variables

### PR DESCRIPTION
**Pull request by bpbond**
## Merge only after #647 

If we are going to shift to proportional LUC losses, as in #647 , then we either keep the `f_luc*` approach as an option (but see note about complexity [here](https://github.com/JGCRI/hector/pull/639#issuecomment-1295813020)) or get rid of it. That is, we can either keep or remove the ability of users to custom-specify how LUC removes/adds carbon to the terrestrial land surface.

This PR does the latter, by cherry-picking the changes from #645 into a new branch . 

This has no effect on the output relative to #647 :

diff

### CODE REVIEW

**Comment by bpbond** [comment]
Updated comment above for clarity.

File: NAMESPACE [hunk 1]
```diff
@@ -17,7 +17,6 @@ export(CH3BR_CONSTRAIN)
 export(CH3CCL3_CONSTRAIN)
 export(CH3CL_CONSTRAIN)
 export(CH4_CONSTRAIN)
-export(CO2_CONSTRAIN)
```
Comment (kdorheim) [inline NAMESPACE:20]: I flagged this as an issue with #639 we are losing a lot of R wrapper functions with this commit.

**Review by kdorheim — CHANGES_REQUESTED** [review]
It looks good other than what I think may have been an accidental change in the arc/rcpp_constatnts file that causes lots of the function to be removed from the R wrapper

File: NAMESPACE (hunk 1 above)
Comment (bpbond) [reply NAMESPACE:20]: I don't understand what's happening here.

**Comment by bpbond** [comment]
@kdorheim Thank you for a36399dfc! Something has gotten borked in my roxygen2 setup, not sure what but it was throwing up all kinds of errors when I asked it to rebuild.

**Review by kdorheim — APPROVED** [review]

**Comment by bpbond** [comment]
Merged in latest `v3_dev` changes and confirming that this PR still does not change output:

diff

diff_percent

Updated comparison data in 18022cd1 and re-enabled old-new test.

**Pull request merged**